Modules/DataModel/Sources/PocketCastsDataModel/Private/Managers/AutoAddQueueDataManager.swift

```diff
@@ -95,7 +115,15 @@ public struct AutoAddCandidatesDataManager {
         public let episodeUuid: String
 
         init?(from resultSet: FMResultSet) {
-            let setting = resultSet.int(forColumn: Constants.autoAddSettingColumnName)
+
+            let setting: Int32
+            if FeatureFlag.settingsSync.enabled {
+                let settingsString = resultSet.string(forColumn: Constants.settingsColumnName)
+                let settings: PodcastSettings? = DBUtils.convertData(value: settingsString?.data(using: .utf8))
```

Comment:
This makes me a bit worried, that for every candidate we'd convert the `PodcastSettings`. We have some users with a crazy amount of podcasts, so I wonder the impact this might have.

I was thinking how we could potentially solve that... question: the `AutoAddCandidates` table will still be updated, right? Do we need to update the code here to use the new settings?

What I'm wondering is if it would be possible to keep things here as they were and mainly keep `AutoAddCandidates` in sync with the new setting. The new setting would serve just to sync, but the logic in how it works would be kept the same (using the `AutoAddCandidates` table and the current query).

I think this will probably mitigate the risk of performance hits.

Reply:
> the AutoAddCandidates table will still be updated, right? Do we need to update the code here to use the new settings?

We could continue to use the `autoAddToUpNext` but that would deviate from storing everything in the `settings` property and cause problems when we go to remove the old settings properties in the future.

`AutoAddCandidates` is kept up to date with which podcasts are available to Up Next, but the position for the Up Next queue is fetch from the `SJPodcasts` table.

I did run some performance tests and it's about 5 times slower on a set of 500 podcasts. Although, the fetch time is still about half a second, so I'm not sure if it's premature optimization.

Reply:
Updating this to use `json_extract` gets the `candidates` fetch down to 1.5 times slower over 500 podcasts.